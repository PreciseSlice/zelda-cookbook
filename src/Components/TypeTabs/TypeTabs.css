@import url("https://fonts.googleapis.com/css?family=PT+Serif:700");
#small-zelda-header {
  align-items: center;
  display: flex;
  font-family: 'PT Serif', serif;
  justify-content: flex-end;
  width: 80%; }

nav li {
  background-color: #424640;
  display: inline-block;
  list-style-type: none;
  width: 90px;
  margin: 0 10px;
  border-radius: 5px;
  min-height: 60px; }

nav a {
  white-space: pre-wrap;
  color: #E9E2C6;
  display: block;
  text-align: center;
  padding: 10px 5px; }

span {
  display: block; }
