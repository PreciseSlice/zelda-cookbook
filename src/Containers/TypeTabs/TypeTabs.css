@import url("https://fonts.googleapis.com/css?family=PT+Serif:700");
select {
  -webkit-appearance: none;
  font-family: fantasy; }

span {
  display: block; }
