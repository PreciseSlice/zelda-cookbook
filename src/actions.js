// Recipes =================

export const populateRecipes = (recipes) => ({
  type: 'POPULATE_RECIPES',
  recipes
})

// Ingredients =============

export const populateIngredients = (ingredients) => ({
  type: 'POPULATE_INGREDIENTS',
  ingredients
})