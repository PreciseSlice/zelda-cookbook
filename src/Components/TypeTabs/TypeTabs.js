import React, { Component } from 'react';
import { connect } from 'react-redux';
import { populateRecipes } from '../../actions'
// import PropTypes from 'prop-types';
import './TypeTabs.css';

export class TypeTabs extends Component {
  constructor() {
    super()
  }

  tabClicked = (e) => {
    const effect = e.target.id.replace('-', ' ')
    const allRecipes = this.props.recipes
    
    const matchingRecipes = allRecipes.filter(recipe => {
      let lowerCaseType = recipe.type.toLowerCase()
      return lowerCaseType.includes(effect)
    })
    this.props.populateRecipes(matchingRecipes)
  }

  render() {
    return (
      <nav onClick={(e) => this.tabClicked(e)} >
        <ul>
        <li><a id="restore-hearts">Restore Hearts</a></li>
        <li><a id="stamina">Restore Stamina</a></li>
        <li><a id="cold">Cold Resist</a></li>
        <li><a id="heat">Heat Resist</a></li>
        <li><a id="electric">Electric Resist</a></li>
        <li><a id="speed">Movement Speed</a></li>
        <li><a id="temp">Temp Max Hearts</a></li>
        <li><a id="defense">Defense Boost</a></li>
        <li><a id="attack">Attack Power</a></li>
        <li><a id="stealth">Stealth Recipes</a></li>
        <li><a id="elixer">Elixer Recipes</a></li>
        </ul>
      </nav>
    )
  } 
}

export const MSTP = ({ recipes }) => ({
  recipes
})

export const MDTP = dispatch => ({
  populateRecipes: matchingRecipes => dispatch(populateRecipes(matchingRecipes))
})

export default connect(MSTP, MDTP)(TypeTabs)