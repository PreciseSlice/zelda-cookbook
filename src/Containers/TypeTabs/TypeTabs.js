import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addTypeFilter } from '../../actions'
import PropTypes from 'prop-types';
import './TypeTabs.css';

export class TypeTabs extends Component {
  constructor(props) {
    super(props);
  }

  selectedType = (event) => {
    const effect = event.target.value
    this.props.addTypeFilter(effect)
  }

  render() {
    return (
      <select onChange={ this.selectedType } id="dropDown">
        <option>Choose Recipe Type</option>
        <option>Restore Hearts</option>
        <option>Restore Stamina</option>
        <option>Cold Resist</option>
        <option>Heat Resist</option>
        <option>Electric Resist</option>
        <option>Movement Speed</option>
        <option>Temporary Max Hearts</option>
        <option>Defense Boost</option>
        <option>Attack Power</option>
        <option>Stealth</option>
        <option>Elixers</option>
      </select>
    )
  } 
}

const { func } = PropTypes;

TypeTabs.propTypes = {
  addTypeFilter: func
};

export const MDTP = dispatch => ({
  addTypeFilter: effect => dispatch(addTypeFilter(effect))  
})

export default connect(null, MDTP)(TypeTabs)