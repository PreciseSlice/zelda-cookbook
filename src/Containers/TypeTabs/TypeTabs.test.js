/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { TypeTabs, MDTP } from './TypeTabs';
import { addTypeFilter } from '../../actions';

describe('TypeTabs', () => {
  let wrapper;
  let mockAddTypeFilter = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<TypeTabs 
      addTypeFilter = { mockAddTypeFilter }
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('selectedType', () => {
    it('should call addTypeFilter with the selected Type of recipe', () => {
      const mockEvent = { target: { value: 'Restore Hearts'}};
      wrapper.instance().selectedType(mockEvent);

      expect(mockAddTypeFilter).toHaveBeenCalledWith('Restore Hearts');
    });
  });

  describe('MDTP', () => {
    it('should call dispatch when addTypeFilter is called', () => {
      const mockDispatch = jest.fn();
      const mockType = 'Restore Hearts';
      const result = MDTP(mockDispatch)

      result.addTypeFilter(mockType);
      expect(mockDispatch).toHaveBeenCalledWith(addTypeFilter(mockType));
    })
  })
});
