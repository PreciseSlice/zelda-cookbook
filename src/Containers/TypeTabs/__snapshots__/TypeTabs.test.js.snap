// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeTabs should match the snapshot 1`] = `
<select
  id="dropDown"
  onChange={[Function]}
>
  <option>
    Choose Recipe Type
  </option>
  <option>
    Restore Hearts
  </option>
  <option>
    Restore Stamina
  </option>
  <option>
    Cold Resist
  </option>
  <option>
    Heat Resist
  </option>
  <option>
    Electric Resist
  </option>
  <option>
    Movement Speed
  </option>
  <option>
    Temporary Max Hearts
  </option>
  <option>
    Defense Boost
  </option>
  <option>
    Attack Power
  </option>
  <option>
    Stealth
  </option>
  <option>
    Elixers
  </option>
</select>
`;
